explainer.md

```diff
@@ -0,0 +1,259 @@
+```js
+requestAnimationFrame(() => {
+  let pendingTransition = document.documentTransition.getPendingTransition();
+  if (pendingTransition.getData().version !== 123)
+    return;
+  // Provides the set of shared-ids for shared elements offered by the outgoing page.
+  if (pendingTransition.offeredSharedElements().has("header"))
+    document.querySelector(".foo").shared-id="header";
+});
+```
+
+Developers can now customize any animation on the Transition DOM. This state is retained until there is an active animation on any pseudo element.
+
+## Transition DOM Details
+For each shared element we create 2 type of pseudo elements that can be targeted by the developer : a container element for the shared element's computed properties (the viewport space transform and border-box size) and a replaced element displaying the shared element's painted content.
+
+One of the motivations behind this split is to provide a stacking context to cross-fade the painted content of incoming and outgoing shared elements. This was necessary to ensure blending identical pixels is a no-op using [plus-lighter](https://drafts.fxtf.org/compositing/#porterduffcompositingoperators_plus_lighter) blending. While same-origin transitions could avoid this, it enables future extensibility for cross-origin transitions where cross-fading identical images would be common.
+
+The example below shows the setup of pseudo elements for a navigation sequence :
+
+* Let's take the following example of an outgoing document before the transition trigger.
+```
+HTML (root)
+├───body
+    ├───div "shared-id=A"
+    ├───div "shared-id=B"
+```
+
+* During the prepare phase, the Transition DOM added to the top-layer is created as follows. shared-container(#shared-id) refers to the container element while shared-old(#shared-id) is the replaced element for the painted content. "root" is a special keyword for the root element. The properties on the shared-container come from the live [computed properties](#computed-properties) of the corresponding shared element.
+```
+top-layer
+├───shared-old(root)
+├───shared-container(#A)
+├        ├───shared-old(#A)
+├───shared-container(#B)
+         ├───shared-old(#B)
+```
+
+* The top-layer state is copied to the incoming Document while its loading. The properties on the shared-container come from the cached [computed properties](#computed-properties) of the shared element in the outgoing Document. The following is an example of the incoming document when it is ready for first render.
+```
+HTML (root)
+├───body
+    ├───div "shared-id=A"
+    ├───div "shared-id=C"
+```
+
+* After ready for first render, the top layer is populated with the Transition DOM for the incoming Document. The properties on the shared-container flip to the live [computed properties](#computed-properties) of the shared element in the incoming Document.
+```
+top-layer
```

Comment:
You could use this kind of diagram, or maybe just HTML?

<top-layer>
  <shared-new>
 ...

Reply:
HTML is good enough. Done.